addons/website_sale/static/src/js/website_sale_tour_shop_cart_recovery.js

```diff
@@ -0,0 +1,104 @@
+odoo.define('website_sale.tour_shop_cart_recovery', function (require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+    var localStorage = require('web.local_storage');
+    var rpc = require('web.rpc');
+    var tour = require('web_tour.tour');
+
+    var productName = "Acoustic Bloc Screens";
+    var orderIdKey = 'website_sale.tour_shop_cart_recovery.orderId';
+    var recoveryLinkKey = 'website_sale.tour_shop_cart_recovery.recoveryLink';
+
+    tour.register('shop_cart_recovery', {
+        test: true,
+        url: '/shop?search=' + productName,
+        wait_for: base.ready(),
+    },
+    [
+        {
+            content: "select " + productName,
+            trigger: '.oe_product_cart a:containsExact("' + productName + '")',
+        },
+        {
+            content: "click add to cart",
+            trigger: '#product_details #add_to_cart',
+        },
+        {
+            content: "check product is in cart, and get cart id",
+            trigger: 'td.td-product_name:contains(' + productName + ')',
+            run: function () {
+                rpc.query({
+                    'model': 'website',
+                    'method': 'sale_get_order',
+                    'args': [[$('html').data('website-id')]],
+                }).then(function (data) {
+                    var orderId = parseInt(data.match(/sale\.order\((\d+),\)/)[1]);
+                    localStorage.setItem(orderIdKey, orderId);
+                    $('body').append('<div id="shop_cart_recovery_rpc_done"></div>');
+                });
+            },
+        },
+        {
+            content: "wait for rpc to be done and logout",
+            trigger: '#shop_cart_recovery_rpc_done',
+            run: function () {
+                window.location.href = "/web/session/logout?redirect=/";
```

Comment:
Maybe avoid that step by doing the redirection in the `then` above and adding an extra trigger on the next step ?

Reply:
Good idea, don't even need an extra trigger because the "login" trigger is enough (the other trigger was for the end of `then`.